roles/openshift_logging_elasticsearch/tasks/main.yaml

```diff
@@ -290,6 +290,24 @@
       path: "{{ generated_certs_dir }}/passwd.yml"
 
 # services
+# bz1636248 - need to delete the service if it exists because clusterIP is immutable
+- name: Check to see if logging-{{ es_component }}-cluster service
+  oc_obj:
+    state: list
+    kind: service
+    name: "logging-{{ es_component }}-cluster"
+    namespace: "{{ openshift_logging_elasticsearch_namespace }}"
+  register: logging_elasticsearch_cluster_service
+
+- name: Remove logging-{{ es_component }}-cluster service
+  oc_service:
+    state: absent
+    name: "logging-{{ es_component }}-cluster"
+    namespace: "{{ openshift_logging_elasticsearch_namespace }}"
+  when:
+  - logging_elasticsearch_cluster_service.results.stderr is not defined
```

Comment:
`logging_elasticsearch_cluster_service.rc == 0` - it is possible for stderr to be empty when there is an error

Reply:
it would always be `rc == 0` unless the module failed, that was a problem i've seen with a lot of places where we are checking if an OCP object exists before we do something. the presence of `stderr` was the most reliable indication since that would be sourced from OCP

Reply:
im not sure about all of the modules, but it is definitely the case with `oc_obj` and `state: list`